### PR TITLE
Compilation Error Fixes

### PR DESCRIPTION
**Pull request by rogerz74**
This PR fixes up the remaining compilation errors we had and fixes the return code problem for `pickMove()` and `gameInstance()` in `human.cc` and `gameRun.cc` respectively

### CODE REVIEW

File: gamerun.cc [hunk 1]
```diff
@@ -24,10 +24,12 @@ int gameRun(Observer *white, Observer *black, ChessGame *gameBoard) {
                     toReturn = 0;               //no legal moves remaining
                     cout << "No move available due to Stalemate!" << endl;
                 }
-            } else {                            //status is just 1
+            } else if (status == 1) {                            //status is just 1
                 white->notify();
                 gameBoard->switchTurn();
                 turn = gameBoard->getPlayerTurn();
+            } else {
+                return -9999; //return code for incomplete game
```
Comment (s32k) [inline gamerun.cc:32]: Make this `toReturn = -9999` since the program stops anyway after that and returns it.

File: gamerun.cc [hunk 2]
```diff
@@ -42,10 +44,12 @@ int gameRun(Observer *white, Observer *black, ChessGame *gameBoard) {
                     toReturn = 0;               //stalemate
                     cout << "No move available due to Stalemate!" << endl;
                 }
-            } else {
+            } else if (status == 1) {
                 black->notify();
                 gameBoard->switchTurn();
                 turn = gameBoard->getPlayerTurn();
+            } else {
+                return -9999; //return code for incomplete game
```
Comment (s32k) [inline gamerun.cc:52]: Same as what I said above.

File: level1.cc [hunk 3]
```diff
@@ -74,22 +74,22 @@ int Level1::pickMove() {
     std::random_device dev;
     std::mt19937 rng(dev());
     std::uniform_int_distribution<std::mt19937::result_type> randompick(0, size-1); //picks random index in newMap
-    std::map<Piece *, Box> iterator item = newMap.begin();                //iterator points to {Piece*:Box}
+    std::map<Piece *, Box>::iterator item = newMap.begin();                //iterator points to {Piece*:Box}
```
Comment (s32k) [inline level1.cc:77]: Is adding the `::` a change that needs to be made in creating each iterator that goes through a map?

Comment (rogerz74) [reply level1.cc:77]: yeah `std::map<Piece *, Box> iterator item` is improper notation.

**Pull request merged**